st2auth_ldap_backend/ldap_backend.py

```diff
@@ -160,15 +164,52 @@ def _ldap_search(self, connection, username, criteria):
                          (base_dn, search_filter, scope, pattern)
         :type criteria: ``dict``
         """
+        results = []
         base_dn = criteria['base_dn']
         search_filter = criteria['search_filter'].format(username=username)
         scope = self._scope_to_ldap_option(criteria['scope'])
 
         LOG.debug('Searching ... %s %s %s' % (base_dn, scope, search_filter))
-        result = connection.search_s(base_dn, scope, search_filter)
+
+        result_id = connection.search(base_dn, scope, search_filter)
+        while 1:
```

Comment:
I've also seen mention of using `search_s` to avoid the loop, but I'd need to do way more digging to know if it's usable as a drop-in replacement for the query being done here.